[dlathyun] iP

dlathyun opened the pull request:
# Kora
> English speaking Korean chatbot in Singapore!
## About
Kora is..
- text-based
- easy to learn and use
- **VERY** FAST 😎 
## Features
- [x] Add tasks with dates!
- [x] Delete tasks
- [x] Find tasks
- [ ] Look nice!
## How to use?
1. download it from [this page](https://github.com/dlathyun/ip)
2. Start using!
## Additional Information
You can also use this to practice Java programming! Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
![Have fun using Kora!](https://img.freepik.com/premium-vector/tiger-character-hanbok_279539-11.jpg)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

dlathyun committed:
Level 01

dlathyun committed:
Level 2

dlathyun committed:
Level 3

dlathyun committed:
Level 4

dlathyun committed:
A-TextUiTesting

dlathyun committed:
Level 5

dlathyun committed:
Level 6

dlathyun committed:
A-Enums

dlathyun committed:
A-TextUiTesting

dlathyun committed:
Level 7

dlathyun committed:
Level 8

dlathyun committed:
Merge branch 'branch-Level-8'

dlathyun committed:
A-MoreOOP

dlathyun committed:
Merge branch 'branch-A-MoreOOP'

dlathyun committed:
Merge remote-tracking branch 'origin/add-gradle-support'

dlathyun committed:
Add package and gradle

dlathyun committed:
Add JUnit tests

dlathyun committed:
Merge branch 'branch-A-JUnit'

dlathyun committed:
A-JavaDoc

dlathyun committed:
A-CodingStandard

dlathyun committed:
Level-9

dlathyun committed:
Merge branch 'branch-A-JavaDoc'

dlathyun committed:
Merge conflict

dlathyun committed:
Merge branch 'branch-Level-9'

dlathyun committed:
Level-9

dlathyun committed:
A-CheckStyle

dlathyun committed:
Merge branch 'branch-A-CheckStyle'

dlathyun committed:
Add Javafx until part 3

dlathyun committed:
Add Javafx until part 4

dlathyun committed:
Add Javafx with greet message

dlathyun committed:
Merge branch 'branch-Level-10'

dlathyun committed:
Add css

dlathyun committed:
Merge branch 'branch-Level-10'

Tang-Moyan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,54 @@
+package duke.parser;
+
+import duke.command.*;
```
It is a good practice that imported classes should always be listed explicitly as according to the java coding standard.

dlathyun replied:
Thanks for pointing it out!

Tang-Moyan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,108 @@
+     * @param info String that contains details of the task.
+     */
+    public Task(String info) {
+        //num = index;
```
Can do a final code cleaning to remove the irrelevant comments, but I guess it is ok to have them at developing stage :) .

Tang-Moyan commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,58 @@
+    /**
+     * Prints greeting message.
+     */
+    public void greet() {
+        String logo = "  _   _   _\n"
+                + " / \\ / \\ / \\\n"
+                + "( 안 | 녕 )\n"
+                + " \\_/ \\_/ \\_/\n";
+        System.out.println(logo + "Hello, I am your chatbot Kora!\nHow can I help you today?");
+        System.out.println("------------------------------");
+    }
+
+    public String getGreetMessage() {
+        String logo = "  _   _   _\n"
+                + " / \\ / \\ / \\\n"
+                + "( 안 | 녕 )\n"
+                + " \\_/ \\_/ \\_/\n";
+        String message = "Hello, I am your chatbot Kora!\nHow can I help you today?";
+        return logo + message;
+    }
```
I think it is possible to reduce these two functions into one. For example, when you want to call greet(), you can just call `System.out.println(getGreetMessage)` instead and then print a line if needed.

dlathyun replied:
Good point haha. Thanks!

Tang-Moyan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,54 @@
+     * @throws KoraException For invalid input.
+     */
+    public static Command parse(String userInput) throws KoraException {
+        String line = "------------------------------" + "\n";
```
I cannot see where this variable is used... In addition, do you think we should make this "line" consistent with the "line" you declared in the Command.java? I personally think it is a good practice to make them refer to a single variable. In this way, you only need to modify at one place to change your display instead of modifying every occurrence :) .

dlathyun replied:
Okay thanks! I think that will be better :)

Tang-Moyan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,54 @@
+        String[] userInputArray = userInput.split("/");
+        Command command;
+        try {
+            if (userInputArray[0].contains("bye")) {
+                command = new ByeCommand();
+            } else if (userInputArray[0].contains("list")) {
+                command = new ListCommand();
+            } else if (userInputArray[0].contains("unmark")) {
+                command = new UnmarkCommand(userInputArray);
+            } else if (userInputArray[0].contains("mark")) {
+                command = new MarkCommand(userInputArray);
+            } else if (userInputArray[0].contains("deadline")) {
+                command = new DeadlineCommand(userInputArray);
+            } else if (userInputArray[0].contains("event")) {
+                command = new EventCommand(userInputArray);
+            } else if (userInputArray[0].contains("todo")) {
+                command = new ToDoCommand(userInputArray);
+            } else if (userInputArray[0].contains("delete")) {
+                command = new DeleteCommand(userInputArray);
+            } else if (userInputArray[0].contains("find")) {
+                command = new FindCommand(userInputArray);
```
The `userInputArray[0].contains("find")` will return true as long as it has a substring "find" in it. In this case, "asdfacads**find**adfaefedcef" will also be considered as a valid command. Similar for the others. What do you think?

dlathyun replied:
Good point.. I will take note. Thanks!

Tang-Moyan reviewed (commented):
Generally very good in following java coding standard which I should give comments on only, as you are my first PR review :) .

simbayippy commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Override
+    public boolean isExitYet() {
```
can naming convention be improved? such as just isExit()

dlathyun replied:
Okay! Will change. Thanks :)

simbayippy commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,29 @@
+    private int taskIndex;
+
+    public MarkCommand(String[] details) {
+        taskIndex = Integer.valueOf(details[0].replace("mark ", ""));
```
what if command is not an Integer value? such as `mark a/ mark $`

dlathyun replied:
Thanks for pointing out. I will take note!

simbayippy reviewed (commented):
Overall very good code quality, could not find much issues with naming convention, and code looks clear and succinct. Maybe in the future could expand on the Exceptions, throwing specific exceptions for specific cases - but still overall really well done!

dlathyun committed:
Add assertions

dlathyun committed:
Add changes after seeing peer review comments

dlathyun committed:
Add changes

dlathyun committed:
Merge pull request #1 from dlathyun/branch-A-Assertions

Add assertions

dlathyun committed:
Add changes

dlathyun committed:
Merge branch 'branch-A-CodeQuality'

dlathyun committed:
Add CI

dlathyun committed:
Merge branch 'branch-A-CI'

dlathyun committed:
Add DateTimeParser

Added DateTimeParser class to handle parsing and formatting
the LocalDateTime.

This was to provide more abstraction and organise the code in
a neater way.

dlathyun committed:
Add DateTimeParser

Added DateTimeParser class to handle parsing and formatting
the LocalDateTime.

This was to provide more abstraction and organise the code in
a neater way.

dlathyun committed:
Add FriendlierSyntax

Implement new classes: SetCommand and UnsetCommand.
Users can use "set" to define their own aliases.

dlathyun committed:
Merge branch 'branch-C-FriendlierSyntax'

dlathyun committed:
Edit Parser class

Kora bot reply "I do not understand" instead of showing error messages.

dlathyun committed:
Update Duke and MainWindow class

The bot replies with appropriate message when KoraException is thrown.

dlathyun committed:
Add extension FlexibleDataSource

ChangeFile Command and LoadFile Command helps to change/load file content.
They also automatically create the file if it does not exist.

dlathyun committed:
Merge branch 'branch-C-FlexibleDataSource'

dlathyun committed:
Change gradle.yml file

dlathyun committed:
Edit files to improve code quality

dlathyun committed:
Add font to css

dlathyun committed:
Edit FXML such that it shows long messages

dlathyun committed:
Add more Korean personality to Kora bot

Kora greets, says sorry, exclaims in Korean.
But don't worry, Kora speaks English too.

dlathyun committed:
Merge branch 'branch-A-Personality'

dlathyun committed:
Add more error handling statements.

Kora gives error messages when:
- Details are not given for tasks.
- Date(s) are not specified for deadlines and events.
- There is something wrong when handling files

dlathyun committed:
Add Help Command

Kora returns a list of available commands and ways to use them.

dlathyun committed:
Merge branch 'branch-C-Help'

dlathyun committed:
Add command for listing command names.

dlathyun committed:
Add more JUnit test cases for classes.

dlathyun committed:
Merge branch 'branch-A-MoreTesting'

dlathyun committed:
Edit DateTimeParser class such that it will remove white speaces
when it parse the string.

dlathyun committed:
Edit DateTimeParser class such that it will remove white speaces
when it parse the string.

dlathyun committed:
Add test cases for checking whether string with white spaces
will get parsed correctly in DateTimeParser.

dlathyun committed:
Add JavaDocs

dlathyun committed:
Edit code to pass checkstyle test

dlathyun committed:
Update UserGuide

dlathyun committed:
Update UserGuide

dlathyun committed:
Update UserGuide

dlathyun committed:
Edit to pass checkstyle

dlathyun committed:
Update UserGuide

dlathyun committed:
Update UserGuide

dlathyun committed:
Update UserGuide

dlathyun committed:
Update UserGuide

dlathyun committed:
Update UserGuide

dlathyun committed:
Update textUiTesting

dlathyun committed:
Update gradle.yml

dlathyun committed:
Update textUiTest

dlathyun committed:
Update UserGuide
